### PR TITLE
Replace sign, verify, private_to_stark_key

### LINKED ISSUE #930: Implement `private_to_stark_key`, `sign`, `verify`
### Feature Request

Functions are from `starkware.crypto.signature.signature`.

Used in `StarkCurveSigner`, message signatures.

### PR DESCRIPTION
**Pull request by drknzz**
Closes #930

## Introduced changes

- use verify from crypto-cpp-py instead of account calling

##

- [ ] This PR contains breaking changes

### CODE REVIEW

File: starknet_py/net/account/base_account.py [hunk 1]
```diff
@@ -188,7 +188,7 @@ def sign_message(self, typed_data: TypedData) -> List[int]:
         """
 
     @abstractmethod
-    async def verify_message(self, typed_data: TypedData, signature: List[int]) -> bool:
+    def verify_message(self, typed_data: TypedData, signature: List[int]) -> bool:
```
Comment (drknzz) [inline starknet_py/net/account/base_account.py:191]: This is a breaking change, needs a mention in docs

File: docs/migration_guide.rst [hunk 2]
```diff
@@ -8,7 +8,18 @@ Migration guide
 Breaking changes
 ----------------
 
-- :meth:`~starknet_py.hash.utils.pedersen_hash` function implementation has been changed, now uses only `crypto-cpp-py` library implementation.
+.. currentmodule:: starknet_py.net.account.base_account
```
Comment (war-in) [inline docs/migration_guide.rst:11]: I'm not sure if we should use it here. This could lead to hard-to-find problems in the future. 
wdyt @cptartur?

**Review by war-in — APPROVED** [review]

File: docs/migration_guide.rst (hunk 2 above)
Comment (drknzz) [reply docs/migration_guide.rst:11]: Migration guide file is written from the bottom-up, so I think there shouldn't be any problems.

Comment (war-in) [reply docs/migration_guide.rst:11]: Ok, fair, but what are the advantages of this approach?

Comment (drknzz) [reply docs/migration_guide.rst:11]: BaseAccount.verify_message is linked to api section

Comment (war-in) [reply docs/migration_guide.rst:11]: And we could do `starknet_py.net.account.base_account.BaseAccount.verify_message` as always

Comment (drknzz) [reply docs/migration_guide.rst:11]: Then it's either the long path with all the package names you listed or just `verify_message` if you precede the path with a tilde `~`.

Current solution leaves only `BaseAccount.verify_message` with a reference.

**Pull request merged**